egs/librimix/ConvTasNet/eval.py

```diff
@@ -41,11 +53,101 @@
 parser.add_argument(
     "--n_save_ex", type=int, default=10, help="Number of audio examples to save, -1 means all"
 )
+parser.add_argument(
+    "--compute-wer", type=int, default=0, help="Compute WER using ESPNet's pretrained model"
+)
+
+COMPUTE_METRICS = ["si_sdr", "sdr", "sir", "sar", "stoi"]
+ASR_MODEL_PATH = (
+    "Shinji Watanabe/librispeech_asr_train_asr_transformer_e18_raw_bpe_sp_valid.acc.best"
+)
+
+
+def update_compute_metrics(compute_wer, metric_list):
+    if not compute_wer:
+        return metric_list
+    try:
+        from espnet2.bin.asr_inference import Speech2Text
+        from espnet_model_zoo.downloader import ModelDownloader
+    except ModuleNotFoundError:
+        import warnings
+
+        warnings.warn("Couldn't find espnet installation. Continuing without.")
+        return metric_list
+    return metric_list + ["wer"]
+
+
+class MockTracker:
+    def __init__(self, *args, **kwargs):
+        pass
 
-compute_metrics = ["si_sdr", "sdr", "sir", "sar", "stoi"]
+    def __call__(self, *args, **kwargs):
+        return dict()
+
+
+class WERTracker:
+    def __init__(self, model_name, trans_df):
+        from espnet2.bin.asr_inference import Speech2Text
+        from espnet_model_zoo.downloader import ModelDownloader
+
+        self.model_name = model_name
+        d = ModelDownloader()
+        self.asr_model = Speech2Text(**d.download_and_unpack(model_name))
+        self.input_txt_list = []
+        self.output_txt_list = []
+        self.sample_rate = int(d.data_frame[d.data_frame["name"] == model_name]["fs"])
+        self.trans_df = trans_df
+        self.trans_dic = self._df_to_dict(trans_df)
+
+    def __call__(
+        self, *, mix: np.ndarray, est_sources: np.ndarray, sample_rate: int, wav_id: List[str]
+    ):
+        """Compute and store best hypothesis for the mixture and the estimates"""
+        if sample_rate != self.sample_rate:
+            mix, est_sources = self.resample(
+                mix, est_sources, fs_from=sample_rate, fs_to=self.sample_rate
+            )
+        input_wer = output_wer = 0.0
+        # Count the mixture output for each speaker
+        txt = self.predict_hypothesis(mix)
+        for tmp_id in wav_id:
+            input_wer += wer(truth=self.trans_dic[tmp_id], hypothesis=txt) / len(wav_id)
+            self.input_txt_list.append(dict(utt_id=tmp_id, text=txt))
```

Comment:
Compute the WER from mixture's hypothesis to each ground truth. Might not be ideal. IDK how it's done in ESPNet.

Reply:
we should accumulate n_words, sub del and ins for each utterance. 
Remember we need to handle the computation for those for both the speakers ( or N speakers) in case we do not perform speaker adaptation.

Reply:
I see the point of doing that but it seems out of scope of this PR maybe. 
For ins, the word order doesn't matter but for del and sub, it does so that's tricky.